### PR TITLE
positive and negative contribution of the shielding constant

### PR DESCRIPTION
**Pull request by rahulj98**
added a way to calculate the positive and negative contributions to the shielding constant by calculating the shielding contribution for each point.
also, we can now create the jvec.vtu files for each atom in a molecule.

### CODE REVIEW

**Comment by heikef** [comment]
I think we should write out the shielding density as a density and not as a vector file. Writing a vtu file for each atom is the same thing as done in my pull request. 
We should keep the separation in positive and negative contributions. 
There are too many commits. Please squash.

File: src/fgimic/jfield.f90 [hunk 1]
```diff
@@ -367,6 +367,7 @@ subroutine jvector_plots(this,tag)
         else if ( ((trim(this%grid%mode)=='base' .or. trim(this%grid%mode)=='std') .and. this%grid%gauss) &
               .or. trim(this%grid%mode)=='file' ) then ! (base && gauss) || file
           allocate(jval_unstructured(this%grid%npts(1), 3))
+          allocate(this%sigmaval(this%grid%npts(1), 3))
```
Comment (heikef) [inline src/fgimic/jfield.f90:370]: Why did you add this to the type structure for the J tensor? Outside the property routine this entry will not be needed.

File: src/fgimic/jfield.f90 [hunk 2]
```diff
@@ -425,19 +421,29 @@ subroutine jvector_plots(this,tag)
             ! read(GRIDELE, '(3i4)') ncells, dummy, dummy
             read(GRIDELE, *) ncells, dummy, dummy
             ! write(*,*) "ncells ", ncells
-            allocate(cells(4,ncells))
+            allocate( this%grid%cells(4,ncells))
```
Comment (heikef) [inline src/fgimic/jfield.f90:424]: Here the inclusion of the cells makes sense. On the other hand we only need this when reading in grid.1.ele.....

File: src/fgimic/jfield.f90 [hunk 3]
```diff
@@ -425,19 +421,29 @@ subroutine jvector_plots(this,tag)
...
+            deallocate(jval_unstructured)
+            deallocate(this%sigmaval)
+            deallocate( this%grid%cells)
+
           else
             write(*,*) 'not writing a vtu file, because the file grid.1.ele was not found.'
           endif
-          deallocate(jval_unstructured)
+
+          ! Maria: moved this call here since the cells are needed when printing
```
Comment (heikef) [inline src/fgimic/jfield.f90:444]: This comments are not needed later on.

File: src/fgimic/jfield.f90 (hunk 1 above)
Comment (rahulj98) [reply src/fgimic/jfield.f90:370]: I am sorry if this is wrong I have limited knowledge of Fortran. 
The variable sigmaval was defined at line 27 in the data type jfield_t, which was called in the subroutine jvector_plots as 'this' at line 256

**Comment by bast** [comment]
Thanks Rahul! Any way I can help to get this change in?

File: src/fgimic/jfield.f90 (hunk 1 above)
Comment (heikef) [reply src/fgimic/jfield.f90:370]: No need to be sorry. I suggest to avoid adding extra fiels to fix type structures. We need this only in the property routine.

**Comment by heikef** [comment]
@bast - I filed a pull request first. When my pull request is merged Rauhl could put his stuff in. Instead of writing the tensor components as vector information on file we should use a new density type format ("scalar"). This has been done (see my pull request). After the merge Rauhul could add the collection of the positive and negative contributions to sigma.
